src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,1077 @@
+            substatus_file_data = json.load(file_handle)
+
+        message = json.loads(substatus_file_data[0]["status"]["substatus"][0]["formattedMessage"]["message"])
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.assertEqual(substatus_file_data[0]["status"]["substatus"][0]["name"], patch_summary)
+        self.assertEqual(substatus_file_data[0]["status"]["substatus"][0]["status"], status.lower())
+        self.assertEqual(len(message["patches"]), patch_count)
+        self.assertNotEqual(message["patches"][-1]['patchId'], "Truncated_patch_list_id")
+        self.assertTrue('Truncated_patch_list_id' not in message["patches"][-1]['name'])
+        self.assertEqual(message["errors"]["code"], 0)
+        self.assertEqual(len(message["errors"]["details"]), 0)
+        self.assertFalse("review this log file on the machine" in message)
+
+    def __assert_patch_summary_from_status(self, substatus_file_data, operation, patch_summary, status, substatus_index=0, is_under_agent_size_limit=True):
+        self.assertEqual(substatus_file_data[0]["status"]["operation"], operation)
+
+        if is_under_agent_size_limit:
+            # Assert status file size < 128kb n 126kb
+            truncated_substatus_file_byte_size = len(json.dumps(substatus_file_data))
+            self.assertTrue(truncated_substatus_file_byte_size < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+            self.assertTrue(truncated_substatus_file_byte_size < Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES)
+        else:
+            # Assert complete status file size > 128kb
+            self.assertTrue(len(json.dumps(substatus_file_data)) > Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+
+        if patch_summary == Constants.PATCH_ASSESSMENT_SUMMARY:
+            substatus_file_data = substatus_file_data[0]["status"]["substatus"][substatus_index]
+        else:
+            substatus_file_data = substatus_file_data[0]["status"]["substatus"][substatus_index]
+
+        self.assertEqual(substatus_file_data["name"], patch_summary)
+        self.assertEqual(substatus_file_data["status"], status.lower())
+
+    def __asert_message_json_from_status(self, substatus_file_data, patch_count, error_count=0):
+        message = json.loads(substatus_file_data["formattedMessage"]["message"])
+        self.assertEqual(patch_count, len(message["patches"]))
+        self.assertEqual(Constants.PatchOperationTopLevelErrorCode.SUCCESS, message["errors"]["code"])
+        self.assertEqual(error_count, len(message["errors"]["details"]))
+
+    def __assert_truncated_error(self, substatus_file_data, error_count):
+        # assert error
+        self.assertEqual(len(json.loads(substatus_file_data["formattedMessage"]["message"])["errors"]["details"]), error_count)
```

Comment:
__asert_message_json_from_status() also validates error_count. There is code duplication here.

Reply:
functionality combined into __assert_patch_summary_from_status, done